javacutil/src/org/checkerframework/javacutil/trees/TreeBuilder.java

```diff
@@ -101,17 +101,19 @@ public MemberSelectTree buildIteratorMethodAccess(ExpressionTree iterableExpr) {
                 (DeclaredType)
                         javacTypes.asSuper((Type) iteratorType, symtab.iteratorType.asElement());
 
-        assert iteratorType.getTypeArguments().size() == 1
-                : "expected exactly one type argument for Iterator";
-
-        TypeMirror elementType = iteratorType.getTypeArguments().get(0);
-        // Remove captured type from a wildcard.
-        if (elementType instanceof Type.CapturedType) {
-            elementType = ((Type.CapturedType) elementType).wildcard;
-
-            iteratorType =
-                    modelTypes.getDeclaredType(
-                            (TypeElement) modelTypes.asElement(iteratorType), elementType);
+        int numIterTypeArgs = iteratorType.getTypeArguments().size();
+        assert numIterTypeArgs <= 1
+                : "expected at most one type argument for Iterator";
+
+        if (numIterTypeArgs == 1) {
+            TypeMirror elementType = iteratorType.getTypeArguments().get(0);
+            // Remove captured type from a wildcard.
+            if (elementType instanceof Type.CapturedType) {
+                elementType = ((Type.CapturedType) elementType).wildcard;
+
+                iteratorType =
+                        modelTypes.getDeclaredType(
+                                (TypeElement) modelTypes.asElement(iteratorType), elementType);
```

Comment:
Not clear to me whether the subsequent `updatedMethodType` code is needed in the case of a raw Iterator type.  But seemed to me it should still work, so I didn't touch it to make this change as narrow as possible